cscs-checks/microbenchmarks/gpu/shmem/shmem.py

```diff
@@ -48,3 +59,40 @@ def __init__(self):
     @sn.sanity_function
     def num_tasks_assigned(self):
         return self.job.num_tasks
+
+    @rfm.run_after('setup')
+    def select_makefile(self):
+        cp = self.current_partition.fullname
+        if cp == 'ault:amdvega':
+            self.prebuild_cmds = ['cp makefile.hip Makefile']
+        else:
+            self.prebuild_cmds = ['cp makefile.cuda Makefile']
+
+    @rfm.run_after('setup')
+    def set_gpu_arch(self):
+        cp = self.current_partition.fullname
+
+        # Deal with the NVIDIA options first
+        nvidia_sm = None
+        if cp in {'tsa:cn', 'ault:intelv100', 'ault:amdv100'}:
+            nvidia_sm = '70'
+        elif cp == 'ault:amda100':
+            nvidia_sm = '80'
+        elif cp in {'dom:gpu', 'daint:gpu'}:
+            nvidia_sm = '60'
+
+        if nvidia_sm:
+            self.build_system.cxxflags += [f'-arch=sm_{nvidia_sm}']
+            if cp in {'dom:gpu', 'daint:gpu'}:
+                self.modules += ['craype-accel-nvidia60']
+            else:
+                self.modules += ['cuda']
+
+        # Deal with the AMD options
+        amd_trgt = None
+        if cp == 'ault:amdvega':
+            amd_trgt = 'gfx906,gfx908'
```

Comment:
Question out of my ignorance: why do you need two targets?

Reply:
Oh, there used to be a node in the `amdvega` partition with different AMD cards than the rest of the nodes. So that's just the product of me messing about with the target and forgetting to cleanup after Mark took that node to other partition.